### DISCUSSION TITLE
Public APIs could be used to populate  people names list

### DISCUSSION THREAD

**Discussion by ccoVeille** [streetsidesoftware/cspell-dicts#3327] [Ideas]
Some APIs could be a good source of candidates for [people-name dictionary](https://github.com/streetsidesoftware/cspell-dicts/tree/main/dictionaries%2Fpeople-names)

**Comment by ccoVeille** [comment]
I also found

https://github.com/RandomAPI/Randomuser.me-Node

This repository could be a gold mine

https://github.com/RandomAPI/Randomuser.me-Node/tree/master/api%2F.nextRelease%2Fdata

**Reply by ccoVeille** [reply]
This one's license is MIT

**Comment by ccoVeille** [comment]
Name day is a good source of candidates for people-name dictionary

https://nameday.abalin.net/docs/#

MIT license, dump provided and available in the repository

**Reply by ccoVeille** [reply]
JSON export has been provided

With https://github.com/xnekv03/nameday-api/commit/fdd39bd67e6a344148400e7780b156dd8ac385cb

Via https://github.com/xnekv03/nameday-api/issues/85

Message above has been updated

**Comment by Jason3S** [comment]
@ccoVeille,

I'm ok with use publicly available data to populate dictionaries. We just need to be very careful of license issues. I would like to bundle the name dictionary with CSpell. But, it cannot be a GPL type of license. That would prevent it from being installed.

**Reply by ccoVeille** [reply]
Indeed. I'm not license-aware enough

**Reply by ccoVeille** [reply]
I opened
- https://github.com/xnekv03/nameday-api/issues/85#issue-2368303230

**Comment by Jason3S** [comment]
The US Census has a list of names. This site refers to it: https://www.mynameday.com/uscensus.html

**Reply by ccoVeille** [reply]
I'm curious about the license, but yes, it's a good list.

I'm impressed by the variety of firstnames for the women

**Reply by Jason3S** [reply]
Here is a cool link: https://www.census.gov/topics/population/genealogy/data.html

US Census data is public by default.

**Reply by ccoVeille** [reply]
Ah OK, indeed. US gov things or gov data

We do have

https://data.gouv.fr here

**Reply by ccoVeille** [reply]
So here you can get all French firstname 

https://www.data.gouv.fr/fr/datasets/fichier-des-prenoms-depuis-1900/

Here is the list of dead people, it includes names I think

https://www.data.gouv.fr/fr/datasets/fichier-des-personnes-decedees/

All these are public
